elide-core/src/main/java/com/yahoo/elide/utils/coerce/CoerceUtil.java

```diff
@@ -76,13 +89,11 @@ private static void setup() {
             public Converter lookup(Class<?> sourceType, Class<?> targetType) {
                 if (targetType.isEnum()) {
                     return TO_ENUM_CONVERTER;
-                } else if (targetType == UUID.class) {
-                    return TO_UUID_CONVERTER;
-                }
-                else if (Map.class.isAssignableFrom(sourceType)) {
+                } else if (Map.class.isAssignableFrom(sourceType)) {
                     return FROM_MAP_CONVERTER;
                 } else if ((String.class.isAssignableFrom(sourceType) || Number.class.isAssignableFrom(sourceType))
-                        && ClassUtils.isAssignable(targetType, Date.class)) {
+                        && ClassUtils.isAssignable(targetType, Date.class)
```

Comment:
It seems like we should be able to handle this in the lookup. Should our lookup map be keyed by a ‘(source, target)’ tuple (null target could imply a default mapping)? This way we wouldn’t have to be smart and the user could define the mapping from ‘String -> Date’ or (ideally) ‘X -> Y’. 

This would be a larger refactor than what we’re probably going for in here, but maybe something to think about.